docs/uml/puml/AuthorListClass/AuthorListClass.puml

```diff
@@ -0,0 +1,46 @@
+@startuml
+!include ../.PlantUMLClass.config
+
+skinparam nodesep 150
+
+class AuthorList {
+    + hasAuthor(author: Author): boolean
+    + hasAuthor(authorName: String): boolean
+    + getAuthor(author: Author): Author
+    + getAuthor(authorName: String): Author
+}
+
+class ArrayList<T>
+ArrayList <|-- AuthorList
+note on link : ArrayList<Author>
+ArrayList <|-- MangaList
+note on link : ArrayList<Manga>
```

Comment:
![Screenshot 2024-10-30 132426](https://github.com/user-attachments/assets/991bbc44-a203-415b-8f97-c46f2f15afc5)
In this diagram, MangaList and AuthorList are indicated as inheriting from ArrayList, but with notes stating "ArrayList<Manga>" and "ArrayList<Author>". I think that, unless these two classes inherit from ArrayList<T> (in which I think you should remove the notes), I believe that there is no inheritance to indicate?

Reply:
Sorry, I'm not sure I understand you. They both inherit from `ArrayList`, but `AuthorList` extends `ArrayList<Author>`, while `MangaList` extends `ArrayList<Manga>`. I still do believe the notes are necessary for show that distinction.

Reply:
Will be keeping this representation based on: 

> Your notation seems OK (* good!). Not sure what the comment means specifically.
> 
> From UML specification: 
> ![image](https://github.com/user-attachments/assets/29424410-ceb6-45c7-bc60-db6798217647)
> 
> *we don't cover this in CS2113; so you went beyond to learn the notation for generics inheritance. 
> specifying the type in the UML note seems OK to me as a way to explain to the reader what it means. 

_Originally posted by Prof in https://github.com/nus-cs2113-AY2425S1/forum/issues/41#issuecomment-2453045543_